Document the `Content-Seed` header (#3)
0x2b3bfa0/minroud: README.md

## Usage

### With [`cml comment`](https://cml.dev/doc/ref/comment)

```console
$ cml comment create --publish --publish-url={endpoint}
```

### With `curl`
```console
$ curl {endpoint} --data-binary @image.png --header "Content-Type: image/png"
https://{bucket}.s3.{region}.amazonaws.com/{hash}
```

#### Overwriting files

> Storage is [content-addressable](https://en.wikipedia.org/wiki/Content-addressable_storage) by default, naming objects after their contents' SHA-256 hash. To return a static/unchanging URL (even after overwriting the contents of the object) provide a `Content-Seed` header; objects will be named after the header's value SHA-256 instead.

Reviewer: ```suggestion
Storage is [content-addressable](https://en.wikipedia.org/wiki/Content-addressable_storage) by default, naming objects after their contents' SHA-256 hash. To return a static/unchanging URL (even after overwriting an object's contents) provide a `Content-Seed` header; objects will be named after this header's SHA-256 hash instead.
```

> Storage is [content-addressable](https://en.wikipedia.org/wiki/Content-addressable_storage) by default, naming objects after their contents' SHA-256 hash. This behavior can be overriden by means of the `Content-Seed` header, whose value will be used to calculate the mentioned hash, so the returned URL remains unchanged after overwriting the contents of the object.

Reviewer: ```suggestion
Storage is [content-addressable](https://en.wikipedia.org/wiki/Content-addressable_storage) by default, naming objects after their contents' SHA-256 hash. To return a static/unchanging URL (even after overwriting the contents of the object) provide a `Content-Seed` header; objects will be named after the header's value SHA-256 instead.
```

```console
$ curl {endpoint} --data-binary @file_v1 --header "Content-Seed: $(uuidgen)"
https://{bucket}.s3.{region}.amazonaws.com/8da7...2297
$ curl {endpoint} --data-binary @file_v2 --header "$_"
https://{bucket}.s3.{region}.amazonaws.com/8da7...2297
```

> **Warning**

> > _Always specify a cryptographically secure `Content-Seed` value, lest allow unauthorized users to read or overwrite the uploaded files._

Reviewer: ```suggestion
> _If using `Content-Seed`, use a cryptographically secure value to prevent unauthorized users from reading/overwriting the uploaded files._
```

maybe also link to some definition of "cryptographically secure?"

Author: > _maybe also link to some definition of "cryptographically secure?"_

Anything that is, at least, as good as the `uuidgen` from `util-linux` in the example. Lacking a canonical and irrefutable definition, a healthy `head --bytes=32 /dev/random | base64` suffices.
